### DISCUSSION TITLE
[Feature Request] CSS in JS/TS capability

### DISCUSSION THREAD

**Discussion by DavraYoung** [microsoft/monaco-editor#3802] [Q&A]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

It would be nice, if have css in js capability for typescript and javascript files.

This will allow to edit code for styled-components, emotion css and other.

For example: 

> css`
>
>   display: ${jsVariable.prop};
>
> `

In VS Code there is an [extension](https://marketplace.visualstudio.com/items?itemName=styled-components.vscode-styled-components)that allows this:
![image](https://user-images.githubusercontent.com/33338429/211178014-1ec8648e-08e4-4985-a341-f224ad9aff07.png)

I tried to find all possible ways to do it, but it seems that currently its not possible.

**Comment by tamayika** [comment]
I read extension implementation and found that its completion is provided by [tsserver plugin](https://github.com/styled-components/typescript-styled-plugin).
But there is no plugin feature in monaco editor typescript mode.
